Scripts/BuildEngine.ps1

```diff
@@ -24,151 +24,143 @@
-        $CMakeCacheVariableOverride += " -DGLFW_BUILD_EXAMPLES=OFF"
-        $CMakeCacheVariableOverride += " -DGLFW_INSTALL=OFF"
-    }
+    Write-Host "Building $systemName $architecture $configuration"
 
-    # ASSIMP options
-    #
-    $CMakeCacheVariableOverride += " -DASSIMP_BUILD_TESTS=OFF"
-    $CMakeCacheVariableOverride += " -DASSIMP_INSTALL=OFF"
-    $CMakeCacheVariableOverride += " -DASSIMP_BUILD_SAMPLES=OFF"
-    $CMakeCacheVariableOverride += " -DASSIMP_BUILD_ASSIMP_TOOLS=OFF"
-    
-    if ($systemName -eq "Darwin") {
-        $CMakeCacheVariableOverride += " -DBUILD_FRAMEWORK=ON"
-    }
+    [string]$BuildDirectoryNameExtension = If ($isMultipleConfig) { "MultiConfig" } Else { $configuration }
+    [string]$BuildDirectoryName = "Result." + $systemName + "." + $architecture + "." + $BuildDirectoryNameExtension
+    [string]$buildDirectoryPath = [IO.Path]::Combine($RepoRoot, $BuildDirectoryName)
+    [string]$cMakeCacheVariableOverride = ""
+    [string]$cMakeGenerator = ""
 
-    # STDUUID options
-    #
-    $CMakeCacheVariableOverride += " -DUUID_BUILD_TESTS=OFF"
-    $CMakeCacheVariableOverride += " -DUUID_USING_CXX20_SPAN=ON"
+    # Create build directory
+    if (-Not (Test-Path $buildDirectoryPath)) {
+        mkdir $buildDirectoryPath
+    } 
+
+    # Building CMake arguments switch to System
+    switch ($systemName) {
+        "Windows" { 
+            switch ($VsVersion) {
+                2019 { 
+                    $cMakeGenerator = "-G `"Visual Studio 16 2019`" -A $architecture"
+                }
+                2022 { 
+                    $cMakeGenerator = "-G `"Visual Studio 17 2022`" -A $architecture"
+                }
+                Default {
+                    throw 'This version of Visual Studio is not supported' 
+                }
+            }          
+            $cMakeCacheVariableOverride = $cMakeCacheVariableOverride, 'DCMAKE_CONFIGURATION_TYPES=Debug;Release' -join " -"
+        }
+        "Linux" { 
+            $cMakeGenerator = "-G `"Unix Makefiles`""
+            $cMakeCacheVariableOverride = $cMakeCacheVariableOverride, 'DSDL_STATIC=ON', 'DSDL_SHARED=OFF' -join " -"
 
-    $CMakeArguments = " -S $RepoRoot -B $BuildDirectoryPath $CMakeGenerator $CMakeCacheVariableOverride"
+            # Set Linux build compiler
+            $env:CC = '/usr/bin/gcc-11'
+            $env:CXX = '/usr/bin/g++-11'
+        }
+        "Darwin" { 
+            $cMakeGenerator = "-G `"Xcode`""
+            $cMakeCacheVariableOverride = $cMakeCacheVariableOverride, 'DBUILD_FRAMEWORK=ON' -join " -"
+        }
+        Default {
+            throw 'This system is not supported'
+        }
+    }
 
-    # Set Linux build compiler
-    #
-    if($systemName -eq 'Linux') {
-        $env:CC= '/usr/bin/gcc-11'
-        $env:CXX= '/usr/bin/g++-11'
+    $cMakeOptions = @("DCMAKE_SYSTEM_NAME=$systemName","DCMAKE_BUILD_TYPE=$configuration", "DBUILD_SANDBOX_PROJECTS=ON", "DENTT_INCLUDE_HEADERS=ON")
```

Comment:
The `DBUILD_SANDBOX_PROJECTS` option can be removed, we deleted sandbox projects